### ISSUE TITLE
Accessibility: new web interface has unlabeled links with screen readers

### ISSUE THREAD

**Issue by davidbtc2009** [mastodon/mastodon#19321]
### Steps to reproduce the problem

1. Go to mastodon.social or mastodon.online
2. Enable a screen reader (i used talkback on android
3. Touch the mastodon logo and talkback reads "link
4. Touch the hashtag, local, fedi, and more along the right edge and talkback only reads " link" for each
...

### Expected behaviour

the links should be properly labeled such as "explore" for the hashtag link, "local timeline" "federated timeline" and "more options"

### Actual behaviour

Talkback only reads "link" when these elements are in focus

### Specifications

Frontend issue
Firefox v105
Android 11
